connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroOutputWriterFactory.scala

```diff
@@ -36,7 +36,8 @@ private[sql] class AvroOutputWriterFactory(
     avroSchemaAsJsonString: String,
     positionalFieldMatching: Boolean) extends OutputWriterFactory {
 
-  private lazy val avroSchema = new Schema.Parser().parse(avroSchemaAsJsonString)
+  private lazy val avroSchema =
+    new Schema.Parser().setValidateDefaults(false).parse(avroSchemaAsJsonString)
```

Comment:
I think, we should only disable validation when parsing Avro schemas coming from files already written previously. This allows us to read files written before Avro 1.9. The Avro project also made this change in Avro 1.10.2 https://github.com/apache/avro/pull/966

However here, we are writing a new file. I don't think we should allow writing files with invalid defaults.

Reply:
+1